includes/header-functions.php

```diff
@@ -136,229 +118,101 @@ function get_header_h1_option( $obj ) {
 }
 
 
-function get_nav_markup( $image=true ) {
-	ob_start();
-?>
-	<nav class="navbar navbar-toggleable-md navbar-custom py-2<?php echo $image ? ' py-sm-4 navbar-inverse header-gradient' : ' navbar-inverse bg-inverse-t-3 py-lg-4'; ?>" aria-label="Site navigation">
-		<div class="container">
-			<button class="navbar-toggler ml-auto mr-0 collapsed" type="button" data-toggle="collapse" data-target="#header-menu" aria-controls="header-menu" aria-expanded="false">
-				<span class="navbar-toggler-text">Navigation</span>
-				<span class="navbar-toggler-icon"></span>
-			</button>
-			<?php
-				wp_nav_menu( array(
-					'theme_location'  => 'header-menu',
-					'depth'           => 1,
-					'container'       => 'div',
-					'container_class' => 'collapse navbar-collapse',
-					'container_id'    => 'header-menu',
-					'menu_class'      => 'nav navbar-nav nav-fill',
-					'fallback_cb'     => 'bs4Navwalker::fallback',
-					'walker'          => new bs4Navwalker()
-				) );
-			?>
-		</div>
-	</nav>
-<?php
-	return ob_get_clean();
-}
-
-
 /**
- * Returns markup for page header title + subtitles within headers that use a
- * media background.
- **/
-function get_header_content_title_subtitle( $obj ) {
-	$title            = get_header_title( $obj );
-	$subtitle         = get_header_subtitle( $obj );
-	$h1               = get_header_h1_option( $obj );
-	$title_elem       = ( $h1 === 'title' ) ? 'h1' : 'span';
-	$subtitle_elem    = ( $h1 === 'subtitle' ) ? 'h1' : 'span';
-	$content_position = get_field( 'page_header_content_position', $obj ) ?: 'left';
-	$position_class   = 'text-' . $content_position;
-
-	ob_start();
-
-	if ( $title ):
-?>
-	<div class="header-content-inner align-self-start pt-4 pt-sm-0 align-self-sm-center">
-		<div class="container <?php echo $position_class; ?>">
-			<div class="d-inline-block bg-primary-t-1">
-				<<?php echo $title_elem; ?> class="header-title"><?php echo $title; ?></<?php echo $title_elem; ?>>
-			</div>
-			<?php if ( $subtitle ) : ?>
-			<div class="clearfix"></div>
-			<div class="d-inline-block bg-inverse">
-				<<?php echo $subtitle_elem; ?> class="header-subtitle"><?php echo $subtitle; ?></<?php echo $subtitle_elem; ?>>
-			</div>
-			<?php endif; ?>
-		</div>
-	</div>
-<?php
-	endif;
+ * Returns the type of header to use for the given object.
+ * The value returned will represent an equivalent template part's name.
+ *
+ * @author Jo Dickson
+ * @since 3.10.0
+ * @param mixed $obj A queried object (e.g. WP_Post, WP_Term), or null
+ * @param array $videos Array of video files set to use in the header background for the given $obj
+ * @param array $images Array of image files set to use in the header background for the given $obj
+ * @return string The header type name
+ */
+function get_header_type( $obj, $videos=null, $images=null ) {
+	$header_type = '';
+
+	$videos = ( $videos !== null ) ? $videos : get_header_videos( $obj );
```

Comment:
We should be able to replace these kinds of lines with the php7 shorthand at this point: `videos ?? get_header_videos ($obj )`. We using other php7 features elsewhere in plugins and what not, so we might as well take advantage.